go-controller/pkg/node/node.go

```diff
@@ -675,7 +675,10 @@ func (n *OvnNode) startEgressIPHealthCheckingServer(wg *sync.WaitGroup, mgmtPort
 			return fmt.Errorf("failed start health checking server due to unsettled IPv6: %w", err)
 		}
 	} else {
-		return fmt.Errorf("unable to start health checking server: no mgmt ip")
+		// OvnNode can be started in environments (e.g DPU) that don't have a management IP. In such situations,
```

Comment:
is DPU is the only case u trying to protect against ?
if so u can check in the config for `OvnKubeNode.Mode == types.NodeModeDPU` and use it to set `EgressIPNodeHealthCheckPort` to `0` to skip, while keeping the original check to catch other anomaly ?

Reply:
Good point. In fact, that was my original intent; but then @wizhaoredhat told me this:
```
I believe this won't handle the case where the node is in "Full" mode as opposed to DPU/DPUHost mode. The new feature that Dmytro has added gives the ability for hardware backed mgmt ports in "Full" mode.
```

So I went ahead and decided we might just have to allow all cases when mgmt ip is not configured.

Reply:
I don't think this is right. The first red flag for me is that we should not disable this feature when we configure to use VF as management port. The fix should be in https://github.com/ovn-org/ovn-kubernetes/pull/3251 

I belive the main issue is that https://github.com/ovn-org/ovn-kubernetes/pull/3251  uses the VF representor and it doesn't have an IP. It's not that there is no IP for the management port.

Imagine if we let this PR go in, then whoever is relying on egress ip health check will also need to know about the details of how the management port was configured. I prefer to keep the hard error here. Evidently, the hard error already helped us catch an issue, i.e. the issue with https://github.com/ovn-org/ovn-kubernetes/pull/3251

Reply:
We were wondering if it would be a better solution to get the VF netdev and use the IP address from that instead of the VF representor (where no IP exists). Then we can have this feature even if the mgmt port is backed by a hardware interface.

Reply:
To correct my previous comment, it's not that https://github.com/ovn-org/ovn-kubernetes/pull/3251 has an issue, but it's that https://github.com/ovn-org/ovn-kubernetes/pull/3251 uncovers the issue that was introduced by the egress ip healthcheck is uncovered more easily.

Reply:
@DmytroLinkin  Any ideas on how we can instead use the correct management port here for the DPU and Full mode cases? I am a afraid we might not be entirely familiar with the DPU/Hardware Backed mgmt port code base to put in the proper fix.

cc @zshi-redhat

Reply:
You're right about DPU mode.

Reply:
@DmytroLinkin the more I think of it, the less I get convinced that we should be doing anything different than what is being proposed in this PR. :)
Egress IP feature uses a label in the node annotation that looks like this:
`kubectl label nodes $WORKER_NODE "k8s.ovn.org/egress-assignable="`. That makes it a candidate egress node.
Looking at https://www.openvswitch.org/support/ovscon2022/slides/OVSCONF2022-Sharing-OVN-among-kubernetes-clusters.pdf  is it correct to assume that `ovnkube-node mode: DPU-host` will have no problem finding the mgmtAddress as we have in the current implementation? Assuming that we will always have one or more ovnkube-node running in the host side, then the egress ip requirements are satisfied, and we are done here. Yes?

Reply:
> @DmytroLinkin the more I think of it, the less I get convinced that we should be doing anything different than what is being proposed in this PR. :) Egress IP feature uses a label in the node annotation that looks like this: `kubectl label nodes $WORKER_NODE "k8s.ovn.org/egress-assignable="`. That makes it a candidate egress node. Looking at https://www.openvswitch.org/support/ovscon2022/slides/OVSCONF2022-Sharing-OVN-among-kubernetes-clusters.pdf is it correct to assume that `ovnkube-node mode: DPU-host` will have no problem finding the mgmtAddress as we have in the current implementation? Assuming that we will always have one or more ovnkube-node running in the host side, then the egress ip requirements are satisfied, and we are done here. Yes?

Typo in presentation. ovnkube-node mode DPU with this change will work. DPU-host and full mode with OVS internal port - works even without this change. Full mode with HW backed device will fail. That happens because across `Start()` function used mgmtPortConfig of the representor, not netdevice. However mgmtPortConfig of the netdevice is stored at array. So the ugly solution will be to get this mgmtPortConfig for the needs of health port server check.

Reply:
@flavio-fernandes what is the reason of doing check at the end of node `Start()`?

Reply:
> @flavio-fernandes what is the reason of doing check at the end of node `Start()`?

@DmytroLinkin  There is no particular reason. As long as it gets done somewhere in the init of the ovnkube-node it should not matter. All it does is to start a listen socket.